### PR TITLE
Jkulhanek/colmap dataparser

### PR DESCRIPTION
**Pull request by jkulhanek**
This PR implements a dataparser to load the COLMAP sparse model directly without the need to run ns-process-data.

### CODE REVIEW

File: nerfstudio/data/dataparsers/colmap_dataparser.py [hunk 1]
```diff
@@ -0,0 +1,386 @@
...
+    masks_path: Optional[Path] = None
+    """Path to masks directory. If not set, masks are not loaded."""
+    depths_path: Optional[Path] = None
+    """Path to depth maps directory. If not set, depths are not loaded."""
+    colmap_path: Path = Path("sparse/0")
+    """Path to the colmap reconstruction directory relative to the data path."""
+
+
+class ColmapDataParser(DataParser):
+    """Nerfstudio DatasetParser"""
```
[the remark this answers is not in the document]
Comment (jkulhanek) [reply nerfstudio/data/dataparsers/colmap_dataparser.py]: Can you please take a look now?

**Comment by jkulhanek** [comment]
I think this dataparser can make it easier for users to use nerfstudio as ns-process-data is no longer required. Only the COLMAP model is sufficient. Currently, ns-process-data does three things:
1. create colmap model
2. downscale images
3. export data as a json

If we accept this dataparser we can instead have a script that would only run colmap (for users which don't run it themselves).

This dataparser should be compatible with folders generated using ns-process-data (as they store the COLMAP model at the same paths). I would even go as far as suggest this as the default for nerfacto and other models. Let me know what you think.

**Review by tancik — APPROVED** [review]
LGTM

**Comment by tancik** [comment]
> I think this dataparser can make it easier for users to use nerfstudio as ns-process-data is no longer required. Only the COLMAP model is sufficient. Currently, ns-process-data does three things:
> 
> 1. create colmap model
> 2. downscale images
> 3. export data as a json
> 
> If we accept this dataparser we can instead have a script that would only run colmap (for users which don't run it themselves).
> 
> This dataparser should be compatible with folders generated using ns-process-data (as they store the COLMAP model at the same paths). I would even go as far as suggest this as the default for nerfacto and other models. Let me know what you think.

Wouldn't this conflict with users that use non-colmap ns-process-data pipelines? It seems like many people use metashape or realitycapture.

**Comment by jkulhanek** [comment]
> > I think this dataparser can make it easier for users to use nerfstudio as ns-process-data is no longer required. Only the COLMAP model is sufficient. Currently, ns-process-data does three things:
> > 
> > 1. create colmap model
> > 2. downscale images
> > 3. export data as a json
> > 
> > If we accept this dataparser we can instead have a script that would only run colmap (for users which don't run it themselves).
> > This dataparser should be compatible with folders generated using ns-process-data (as they store the COLMAP model at the same paths). I would even go as far as suggest this as the default for nerfacto and other models. Let me know what you think.
> 
> Wouldn't this conflict with users that use non-colmap ns-process-data pipelines? It seems like many people use metashape or realitycapture.

Yes, let’s keep the default then. I didn’t know many people used non colmap pipelines and it would make it more difficult for them.

**Pull request merged**